vmm/src/cpu.rs

```diff
@@ -1128,7 +1153,8 @@ impl CpuManager {
 
         match desired_vcpus.cmp(&self.present_vcpus()) {
             cmp::Ordering::Greater => {
-                self.create_vcpus(desired_vcpus, None)?;
+                self.create_vcpus(desired_vcpus)?;
+                self.configure_vcpus(None)?;
```

Comment:
You could avoid double configuring of vCPUs by returning a `Vec<Arc<Mutex<Vcpu>>>` of the newly created vCPUs here and only configuring those. This would remove the need to have the `configured` bool.

Reply:
Configuring the correct vCPUs sounds the correct way to move, I will update later today.